Mathlib/Analysis/Meromorphic/Order.lean

```diff
@@ -87,6 +90,115 @@ lemma order_eq_int_iff {n : ℤ} (hf : MeromorphicAt f x) : hf.order = n ↔
+      refine tendsto_nhdsWithin_iff.2 ⟨?_, ?_⟩
+      · have : ContinuousWithinAt (fun z ↦ z - x) ({x}ᶜ) x :=
+          ContinuousAt.continuousWithinAt (by fun_prop)
+        simpa using this.tendsto
+      · filter_upwards [self_mem_nhdsWithin] with y hy
+        simpa [sub_eq_zero] using hy
+    apply Tendsto.comp (NormedField.tendsto_norm_zpow_nhdsNE_zero_atTop m_neg) this
+  apply A.congr'
+  filter_upwards [hg] with z hz using by simp [hz]
+
+/-- If the order of a meromorphic function is zero, then this function converges to a nonzero
+limit at this point. See also the iff version `tendsto_ne_zero_iff_order_eq_zero`. -/
+lemma tendsto_ne_zero_of_order_eq_zero (hf : MeromorphicAt f x) (ho : hf.order = 0) :
+    ∃ c ≠ 0, Tendsto f (𝓝[≠] x) (𝓝 c) := by
+  rcases hf.order_eq_int_iff.1 ho with ⟨g, g_an, gx, hg⟩
+  refine ⟨g x, gx, ?_⟩
+  apply g_an.continuousAt.continuousWithinAt.tendsto.congr'
+  filter_upwards [hg] with y hy using by simp [hy]
+
+/-- If the order of a meromorphic function is positive, then this function converges to zero
+at this point. See also the iff version `tendsto_zero_iff_order_pos`. -/
+lemma tendsto_zero_of_order_pos (hf : MeromorphicAt f x) (ho : 0 < hf.order) :
+    Tendsto f (𝓝[≠] x) (𝓝 0) := by
+  cases h'o : hf.order with
+  | top =>
+    apply tendsto_const_nhds.congr'
+    filter_upwards [hf.order_eq_top_iff.1 h'o] with y hy using hy.symm
+  | coe n =>
+    rcases hf.order_eq_int_iff.1 h'o with ⟨g, g_an, gx, hg⟩
+    lift n to ℕ using by simpa [h'o] using ho.le
+    have : (0 : E) = (x - x) ^ n • g x := by
+      have : 0 < n := by simpa [h'o] using ho
+      simp [zero_pow_eq_zero.2 this.ne']
+    rw [this]
+    have : ContinuousAt (fun z ↦ (z - x) ^ n • g z) x := by fun_prop
+    apply this.continuousWithinAt.tendsto.congr'
+    filter_upwards [hg] with y hy using by simp [hy]
+
+/-- If the order of a meromorphic function is nonnegative, then this function converges
+at this point. See also the iff version `tendsto_nhds_iff_order_nonneg`. -/
+lemma tendsto_nhds_of_order_nonneg (hf : MeromorphicAt f x) (ho : 0 ≤ hf.order) :
+    ∃ c, Tendsto f (𝓝[≠] x) (𝓝 c) := by
+  rcases ho.eq_or_lt with ho | ho
+  · rcases hf.tendsto_ne_zero_of_order_eq_zero ho.symm with ⟨c, -, hc⟩
+    exact ⟨c, hc⟩
+  · exact ⟨0, hf.tendsto_zero_of_order_pos ho⟩
+
+/-- A meromorphic function converges to infinity iff its order is negative. -/
+lemma tendsto_cobounded_iff_order_neg (hf : MeromorphicAt f x) :
+    Tendsto f (𝓝[≠] x) (Bornology.cobounded E) ↔ hf.order < 0 := by
+  rcases lt_or_le hf.order 0 with ho | ho
+  · simp [ho, hf.tendsto_cobounded_of_order_neg]
+  · simp only [lt_iff_not_le, ho, not_true_eq_false, iff_false, ← tendsto_norm_atTop_iff_cobounded]
+    obtain ⟨c, hc⟩ := hf.tendsto_nhds_of_order_nonneg ho
+    exact not_tendsto_atTop_of_tendsto_nhds hc.norm
+
+/-- A meromorphic function converges to a limit iff its is order if nonnegative. -/
+lemma tendsto_nhds_iff_order_nonneg (hf : MeromorphicAt f x) :
```

Comment:
Reading some of these proofs makes me think the de Bruijn factor of formalisation is still too high. But that's not a fault of your proofs.